lib/open_api_spex/schema.ex

```diff
@@ -432,28 +432,42 @@ defmodule OpenApiSpex.Schema do
   def validate(%Schema{type: type}, nil, path, _schemas) do
     {:error, "#{path}: null value where #{type} expected"}
   end
-  def validate(schema = %Schema{type: type}, value, path, _schemas) when type in [:integer, :number] do
+  def validate(schema = %Schema{type: :integer}, value, path, _schemas) when is_integer(value) do
+    # TODO DUPLICATION
+    with :ok <- validate_multiple(schema, value, path),
+         :ok <- validate_maximum(schema, value, path),
+           :ok <- validate_minimum(schema, value, path) do
+      :ok
+    end
+  end
+  def validate(schema = %Schema{type: :number}, value, path, _schemas) when is_number(value) do
+    # TODO DUPLICATION
     with :ok <- validate_multiple(schema, value, path),
          :ok <- validate_maximum(schema, value, path),
          :ok <- validate_minimum(schema, value, path) do
       :ok
     end
   end
-  def validate(schema = %Schema{type: :string}, value, path, _schemas) do
+  def validate(schema = %Schema{type: :string}, value, path, _schemas) when is_binary(value) do
+    # TODO DUPLICATION
     with :ok <- validate_max_length(schema, value, path),
          :ok <- validate_min_length(schema, value, path),
-         :ok <- validate_pattern(schema, value, path),
-         :ok <- validate_enum(schema, value, path) do
+           :ok <- validate_pattern(schema, value, path),
+           :ok <- validate_enum(schema, value, path) do
```

Comment:
Alignment here looks off - feel free to run `mix format` on the file in a separate commit.